packages/ui-checkbox/src/CheckboxGroup/index.tsx

```diff
@@ -59,21 +61,18 @@ class CheckboxGroup extends Component<CheckboxGroupProps> {
     children: null
   }
 
-  // @ts-expect-error ts-migrate(7006) FIXME: Parameter 'props' implicitly has an 'any' type.
-  constructor(props) {
-    // @ts-expect-error ts-migrate(2554) FIXME: Expected 1-2 arguments, but got 0.
-    super()
+  constructor(props: CheckboxGroupProps) {
+    super(props)
 
     if (typeof props.value === 'undefined') {
       this.state = {
-        value: props.defaultValue
+        value: props.defaultValue || []
```

Comment:
If you give this a default value, then a lot of checks are not necessary in the code. Can you please double-check if this has any side-effects? If not, then we can simplify the code a bit.

Reply:
It is only used in one place, so the simplification would only be "|| []"->"!". I think setting the defaultValue prop to a preset default (even if its an empty array) is counterintuitive, since it should be user set or undefined

Reply:
I meant that if the `this.state.value` has a default of `[]`, then it can never be undefined, so there are some unnecessary checks in the code. For example the `get value() {}` getter can be refactored a bit and the first line of `handleChange` too.